### ISSUE TITLE
[Feature request]

### ISSUE THREAD

**Issue by broli** [Uminosachi/sd-webui-inpaint-anything#127]
I want to be able to select a "section" inside an image, before running segment anything, to improve the chances that it detects all the details i want, and to improve speed 

It dosnt have to be precise, you can have a pre made 3x3 grid and let the user choose one or more

I feel it would improve the detection and creation of masks 

PD: thanks for the most important extension ever!

**Comment by Uminosachi** [comment]
Indeed, pre-processing images by grid segmenting them beforehand seems to have several advantages. It not only speeds up the process but also allows us to focus on just the essential parts. However, this approach could slightly complicate the Web UI, so I need to give it some more thought.

**Comment by broli** [comment]
> Indeed, pre-processing images by grid segmenting them beforehand seems to have several advantages. It not only speeds up the process but also allows us to focus on just the essential parts. However, this approach could slightly complicate the Web UI, so I need to give it some more thought.

if you share what kind of "complexity" you are thinking, maybe someone will see the post and have and idea ?
